schemas/pci/pci-bus.yaml

```diff
@@ -22,139 +23,202 @@ description: |
+
+      device_type:
+        const: pci
+
       reg:
         items:
           minItems: 5
           maxItems: 5
         minItems: 1
         maxItems: 6   # Up to 6 BARs
+
+      vendor-id:
+        description: The PCI vendor ID
+        $ref: /schemas/types.yaml#/definitions/uint32
+
+      device-id:
+        description: The PCI device ID
+        $ref: /schemas/types.yaml#/definitions/uint32
+
     required:
+      - device_type
       - reg
 
-required:
-  - device_type
-  - ranges
-  - "#address-cells"
-  - "#size-cells"
+
+  pci-bridge:
+    title: PCI-to-PCI bridge function
+
+    description: |
+      A hardware device that is, from an electrical standpoint, a single
+      PCI function on one PCI bus (the "parent" bus) and the bus controller
+      of a secondary PCI bus (the "child" bus).
+
+    allOf:
+      - $ref: "#/$defs/pci-child-bus"
+      - $ref: "#/$defs/pci-function"
+      - properties:
+          compatible:
+            contains:
+              const: "pciclass,0604"
+          reg:
+            maxItems: 2   # PCI-to-PCI bridge can have up to 2 BARs
+
+
+  pci-child-bus:
+    title: PCI child bus node
+
+    description: |
+      Node which represents segment of PCI bus. For PCI-to-PCI bridge it is
+      secondary bus. Subnodes of this node represents of PCI functions.
+
+    properties:
+      device_type:
+        const: pci
+
+      bus-range:
+        $ref: /schemas/types.yaml#/definitions/uint32-array
+        minItems: 2
+        maxItems: 2
+        items:
+          maximum: 255
+
+      "#interrupt-cells":
+        const: 1
+
+      interrupt-map: true
+#        minItems: 1
+#        maxItems: 88    # 22 IDSEL x 4 IRQs
+#        items:
+#          minItems: 6   # 3 addr cells, 1 PCI IRQ cell, 1 phandle, 1+ parent addr and IRQ cells
+#          maxItems: 16
+
+      interrupt-map-mask:
+        items:
+          - description: PCI high address cell
+            minimum: 0
+            maximum: 0xf800
+          - description: PCI mid address cell
+            const: 0
+          - description: PCI low address cell
+            const: 0
+          - description: PCI IRQ cell
+            minimum: 0
+            maximum: 7
+
+    patternProperties:
+      ".*":
+        allOf:
+          - if:
+              type: object
+              properties:
+                device_type:
+                  const: pci
+            then:
+              allOf:
+                - $ref: "#/$defs/pci-function"
+              properties:
+                $nodename:
+                  pattern: "^[a-zA-Z][a-zA-Z0-9,+\\-._]{0,63}@1?[0-9a-f](,[0-7])?$"
+          - if:
+              type: object
+              properties:
+                compatible:
+                  contains:
+                    const: "pciclass,0604"
+            then:
+              $ref: "#/$defs/pci-bridge"
+
+      "@1?[0-9a-f](,[0-7])?$":
+        $ref: "#/$defs/pci-function"
+
+    required:
+      - device_type
+
+
+  pci-master-bus:
```

Comment:
pci-host-bridge would be a better name IMO. However, we need to keep pci-bus.yaml to avoid breaking everyone.

Reply:
I used _master bus_ because this is how it is called in _PCI Bus Binding to: IEEE Std 1275-1994_ document at: http://www.devicetree.org/open-firmware/bindings/pci/pci2_1.pdf

Reply:
Okay